### ISSUE TITLE
Daemon does not detect blockchain events

### ISSUE THREAD

**Issue by art4melody** [spesmilo/electrum#1023]
I use modified `merchant.py` provided in the sample and create it as a daemon to generate address and 'listen' to blockchain events in the background.
But after running for several days, it does not detect transaction from electrum servers. I have to restart the script -> reconnect to electrum daemon, then the data coming in.
This does not happen only once, but several times already.
Here is the contents of `daemon.log`:
```
new client: 1
blocks: 342983
connected to EAST.electrum.jdubya.info 50002
restarting main interface
connected to jwu42.hopto.org 50002
restarting main interface
connected to electrum.drollette.com 50002
restarting main interface
connected to electrum.hsmiths.com 50002
restarting main interface
connected to electrum.no-ip.org 50002
restarting main interface
connected to us.electrum.be 50002
restarting main interface
switching to us.electrum.be:50002:s
connected to eco-electrum.ddns.net 50002
connected to electrum.bitfuzz.nl 50002
connected to electrum.dragonzone.net 50002
connected to us.electrum.be 50002
connected to us.electrum.be 50002
connected to us.electrum.be 50002
connected to us.electrum.be 50002
connected to us.electrum.be 50002
height: 342984 electrum.dragonzone.net:50002:s
height: 342985 electrum.dragonzone.net:50002:s
height: 342986 electrum.dragonzone.net:50002:s
height: 342987 electrum.dragonzone.net:50002:s
height: 342988 EAST.electrum.jdubya.info:50002:s
height: 342989 electrum.dragonzone.net:50002:s
height: 342990 EAST.electrum.jdubya.info:50002:s
height: 342991 EAST.electrum.jdubya.info:50002:s
height: 342992 jwu42.hopto.org:50002:s
height: 342993 EAST.electrum.jdubya.info:50002:s
height: 342994 electrum.hsmiths.com:50002:s
height: 342995 electrum.dragonzone.net:50002:s
height: 342996 electrum.dragonzone.net:50002:s
height: 342997 electrum.bitfuzz.nl:50002:s
height: 342998 electrum.hsmiths.com:50002:s
height: 342999 EAST.electrum.jdubya.info:50002:s
height: 343000 electrum.dragonzone.net:50002:s
height: 343001 electrum.dragonzone.net:50002:s
height: 343002 electrum.dragonzone.net:50002:s
height: 343003 electrum.dragonzone.net:50002:s
height: 343004 electrum.dragonzone.net:50002:s
height: 343005 electrum.dragonzone.net:50002:s
```
It does not show any errors near the end of the file (where it does not detect payment).

Any clue why this happen? I guess this may have a relation with network.

**Comment by art4melody** [comment]
Today there is a lot of connection issue in the daemon:
```
height: 343121 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343122 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343123 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343124 electrum.thwg.org:50002:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343125 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343126 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343127 electrum.random.re:110:s
forcing reconnection
forcing reconnection
height: 343128 electrum.random.re:110:s
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
forcing reconnection
height: 343129 electrum.random.re:110:s
forcing reconnection
forcing reconnection
new client: 1
client quit: 0
new client: 1
client quit: 0
new client: 1
client quit: 0
```
And I can't connect to the daemon. `network.is_connected()` returns false after connecting.

**Comment by ecdsa** [comment]
should be fixed with yesterday's change: 911a60c02d824b02e698c967420aebe14b937be1
